src/prognostic_equations/closures.jl

```diff
@@ -0,0 +1,94 @@
+#####
+##### Grid-mean closures (mixing length)
+#####
+
+import ClimaCore.Geometry as Geometry
+import ClimaCore.Fields as Fields
+
+"""
+    Compute the Smagorinsky length scale from
+    - c_smag coefficient
+    - N_eff - buoyancy frequency = sqrt(max(ᶜlinear_buoygrad, 0))
+    - dz - vertical grid scale
+    - Pr - Prandtl number
+    - ϵ_st - strain rate norm
+"""
+function smagorinsky_length_scale(c_smag, N_eff, dz, Pr, ϵ_st)
```

Comment:
If we want to be picky (and maybe we do), a better name would be `smagorinsky_lilly_length`.